src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -179,10 +185,22 @@ IOptions<FeatureConfiguration> featureConfiguration
 
             if (_enableDropInvalidDicomJsonMetadata)
             {
-                // drop invalid metadata
+
                 foreach (DicomTag tag in storeValidatorResult.InvalidTagErrors.Keys)
                 {
+                    // drop invalid metadata
                     dicomDataset.Remove(tag);
+
+                    InstanceIdentifier instanceIdentifier = new InstanceIdentifier(
+                        dicomDataset.GetString(DicomTag.StudyInstanceUID),
+                        dicomDataset.GetString(DicomTag.SeriesInstanceUID),
+                        dicomDataset.GetString(DicomTag.SOPInstanceUID));
+                    string message = storeValidatorResult.InvalidTagErrors[tag].Error;
+                    LogForwarder.LogTrace(
+                        _telemetryClient,
+                        $"{message}. This attribute will be dropped from JSON metadata and can not be used to index." +
+                        $" This attribute will remain in the Dicom binary file.",
+                        instanceIdentifier);
```

Comment:
If that was an extension method, then 
```suggestion
                    _telemetryClient.LogTrace(
                        $"{message}. This attribute will be dropped from JSON metadata and can not be used to index." +
                        $" This attribute will remain in the Dicom binary file.",
                        instanceIdentifier);
```